functions/function.go

```diff
@@ -0,0 +1,289 @@
+/*
+Copyright 2022-Present Couchbase, Inc.
+
+Use of this software is governed by the Business Source License included in
+the file licenses/BSL-Couchbase.txt.  As of the Change Date specified in that
+file, in accordance with the Business Source License, use of this software will
+be governed by the Apache License, Version 2.0, included in the file
+licenses/APL2.txt.
+*/
+
+package functions
```

Comment:
I think the main motivation for a separate functions package is code organization/isolation?  If that's accurate, what's your opinion on defining this package as db/functions instead of a root level functions?  I don't think we anticipate any functions that operate outside of the context of a database, so it seems better from an organizational perspective.

It also seems like the main source of import recursion (necessitating the db/functions.go interfaces) is in the db package's blip code - having db/functions might lead us toward a db/blipSync in future that would allow us to break that recursion and remove that interface code.

Reply:
That sounds like a good idea. I'll move the package.